### PR TITLE
Update token guides for Cadence 1.0

### PR DESCRIPTION
**Pull request by joshuahannan**
Updates the ft, nft, and metadata views guides for Cadence 1.0
I'm currently working on updating Chase's example repos, so those should be ready soon.
I'm hoping to get a review on this first though in case there are any changes I need to make to the code

### CODE REVIEW

File: docs/build/guides/fungible-token.md [hunk 1]
```diff
@@ -494,31 +872,34 @@ To learn more about running transactions using CLI, [see the docs](https://devel
 
 Let's now read the balance of the newly created account (`test-acct`) to check it's zero.
 
-Create this new script file `cadence/scripts/ReadVaultBalance.cdc`:
+Create this new script file `cadence/scripts/get_fooToken_balance.cdc`:
```
Comment (sisyphusSmiling) [inline docs/build/guides/fungible-token.md:875]: I sticking with snake case would be more consistent
```suggestion
Create this new script file `cadence/scripts/get_foo_token_balance.cdc`:
```

File: docs/build/guides/fungible-token.md [hunk 2]
```diff
@@ -494,31 +872,34 @@ To learn more about running transactions using CLI, [see the docs](https://devel
...
 
 ```bash
-flow generate script ReadVaultBalance
+flow generate script get_fooToken_balance
```
Comment (sisyphusSmiling) [inline docs/build/guides/fungible-token.md:878]: Follow on from above. Also applies to a couple other places below.

```suggestion
flow generate script get_foo_token_balance
```

File: docs/build/guides/nft.md [hunk 3]
```diff
@@ -153,7 +130,7 @@ access(all) contract FooBar {
 
     init() {
         self.totalSupply = 0
```
Comment (sisyphusSmiling) [inline docs/build/guides/nft.md:132]: This field is removed above, not sure if it should be here or the intention was to remove it

File: docs/build/guides/nft.md [hunk 4]
```diff
@@ -183,223 +160,344 @@ access(all) contract FooBar {
...
+        // ...Collection Code
+    }
+
+    // ...rest of the contract code
+
 }
 ```
+These interface conformances for [NFT](https://github.com/onflow/flow-nft/blob/master/contracts/NonFungibleToken.cdc#L98)
+and [Collection](https://github.com/onflow/flow-nft/blob/master/contracts/NonFungibleToken.cdc#L190)
+inherit from other interfaces provide important functionality and restrictions
```
Comment (sisyphusSmiling) [inline docs/build/guides/nft.md:195]:
```suggestion
inherited from other interfaces provide important functionality and restrictions
```

File: docs/build/guides/nft.md [hunk 5]
```diff
@@ -183,223 +160,344 @@ access(all) contract FooBar {
...
+This is an example of entitlements in Cadence.
+[Entitlements](https://cadence-lang.org/docs/1.0/language/access-control#entitlements)
+are a way for developers to restrict access to privileged fields and functions
+in a composite type like a resource when a reference is created for it.
+In this example, the `withdraw()` function is always accessible to code that
+controls the full `Collection` object, but if a reference is created for it,
+the `withdraw()` function can only be called if the reference 
+is authorized by the owner with `NonFungibleToken.Withdraw`,
+which is [a standard entitlement](https://github.com/onflow/flow-nft/blob/master/contracts/NonFungibleToken.cdc#L58)
+defined by the FungibleToken contract:
```
Comment (sisyphusSmiling) [inline docs/build/guides/nft.md:288]:
```suggestion
defined by the NonFungibleToken contract:
```

File: docs/build/guides/nft.md [hunk 6]
```diff
@@ -473,40 +597,46 @@ flow accounts create
...
+To retrieve the NFTs associated with an account, you'll need a script.
+Scripts are read-only operations that allow you to query the blockchain.
+They don't modify the blockchain's state, and therefore,
+they don't require gas fees or signatures (read more about scripts here).
 
 Start by creating a script file using the `generate` command again:
 
 ```bash
-flow generate script GetNFTs
+flow generate script get_fooBar_ids
```
Comment (sisyphusSmiling) [inline docs/build/guides/nft.md:615]: Similar comment as FT guide, snake case would be more consistent IMO

File: docs/build/guides/nft.md [hunk 7]
```diff
@@ -590,159 +760,28 @@ flow accounts create
...
-    // ...[Collection init]...
-}
-```
+The transfer transaction also has a [generic version](https://github.com/onflow/flow-nft/blob/master/transactions/generic_transfer_with_address.cdc) that developers are encouraged to use!
 
 Congrats, you did it! You’re now ready to launch the next fun NFT project on Flow.
 
 ## More
 
 - Explore [an example NFT repository](https://github.com/nvdtf/flow-nft-scaffold/blob/main/cadence/contracts/exampleNFT/ExampleNFT.cdc)
```
Comment (sisyphusSmiling) [inline docs/build/guides/nft.md:784]: We'll want to make sure this repo is updated as wel

**Review by sisyphusSmiling — APPROVED** [review]
Looks great! Left a couple smaller suggestions, mostly just about consistency and linked repos being updated

File: docs/build/guides/fungible-token.md [hunk 8]
```diff
@@ -42,7 +75,7 @@ If you haven't installed the Flow CLI yet and have [Homebrew](https://brew.sh/)
 
 ### Initializing a New Project
 
-> 💡 Note: Here is [a link to the completed code](https://github.com/chasefleming/foobar-nft) if you want to skip ahead or reference as you follow along.
+> 💡 Note: Here is [a link to the completed code](https://github.com/chasefleming/FooToken) if you want to skip ahead or reference as you follow along.
```
[the remark this answers is not in the document]
Comment (joshuahannan) [reply docs/build/guides/fungible-token.md:78]: I'm working on updating that right now. I'll have a PR ready soon

File: docs/build/guides/fungible-token.md [hunk 9]
```diff
@@ -95,31 +146,45 @@ In this contract file, we want to import our `FungibleToken` contract that we've
...
 access(all) contract FooToken: FungibleToken {
-  access(all) var totalSupply: UFix64
+    access(all) var totalSupply: UFix64
 
-  init() {
-    self.totalSupply = 1000.0
-  }
+    access(all) let VaultStoragePath: StoragePath
+    access(all) let VaultPublicPath: PublicPath
+    access(all) let MinterStoragePath: StoragePath
```
[the remark this answers is not in the document]
Comment (joshuahannan) [reply docs/build/guides/fungible-token.md:162]: yep, also working on that

File: docs/build/guides/fungible-token.md (hunk 1 above)
Comment (joshuahannan) [reply docs/build/guides/fungible-token.md:875]: good idea. I'll update all the txs and scripts to have that

**Comment by joshuahannan** [comment]
@chasefleming I opened PRs in your repos with the changes from these guides:
* NFT: https://github.com/chasefleming/foobar-nft/pull/1
* FT: https://github.com/chasefleming/FooToken/pull/2

Can you please review this PR and those two PRs so we can get everything merged an updated? Thanks!

File: docs/build/guides/fungible-token.md [hunk 10]
```diff
@@ -17,22 +17,55 @@ Fungible tokens are digital assets that are interchangeable and indistinguishabl
...
+fungible tokens are stored in structures called resources.
+Resources are objects in Cadence that store data,
+but have special restrictions about how they can be stored and transferred,
+making them perfect for representing digital objects with real value.
 
-1. A temporary vault (or a temporary piggy bank) is created holding the transfer amount.
+You can learn more about resources in the Cadence [documentation](https://cadence-lang.org/docs/1.0/language/resources)
+and [tutorials](https://cadence-lang.org/docs/1.0/tutorial/resources).
+
+For Fungible Tokens specifically, tokens are represented by a resource type called a `Vault`:
```
Comment (chasefleming) [inline docs/build/guides/fungible-token.md:29]: Should "Fungible Tokens" be capitalized here?

**Review by chasefleming — CHANGES_REQUESTED** [review]
I started noting the sections with line break issues but maybe there is a formatting in your editor to just do it all in one go.

**Comment by joshuahannan** [comment]
@chasefleming I addressed your comments and pushed some new changes. As I said in my reply, the line break thing was intentional, so don't worry about that

**Review by chasefleming — APPROVED** [review]

**Pull request merged**